src/App.tsx

```diff
@@ -15,36 +17,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+function getPrepareGoods(goods: string[], sortBy: string, isReversed: boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [typeSort, setTypeSort] = useState('');
+  const [isReversed, setReverse] = useState(false);
+  const sortedGoods = getPrepareGoods(goodsFromServer, typeSort, isReversed);
+
+  const reset = () => {
+    setTypeSort('');
+    setReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': typeSort !== SortType.Alphabet,
+          })}
+          onClick={() => setTypeSort(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': typeSort !== SortType.Length,
+          })}
+          onClick={() => setTypeSort(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReverse(!isReversed)}
```

Comment:
```suggestion
          onClick={() => setReverse(isReversed => !isReversed)}
```
Always use callback to update state based on previous value